bulk_site_creator/views.py

```diff
@@ -0,0 +1,253 @@
+    job_query_params = {
+        'KeyConditionExpression': Key('pk').eq(school_key) & Key('sk').eq(job_id),
+        'ScanIndexForward': False,
+    }
+    job = table.query(**job_query_params)['Items'][0]
+
+    tasks_query_params = {
+        'KeyConditionExpression': Key('pk').eq(job_id),
+        'ScanIndexForward': False,
+    }
+    tasks = table.query(**tasks_query_params)['Items']
+
+    context = {
+        'job': job,
+        'tasks': tasks
+    }
+    return render(request, "bulk_site_creator/job_detail.html", context=context)
+
+
+@login_required
+@lti_role_required(const.ADMINISTRATOR)
+@lti_permission_required(settings.PERMISSION_BULK_SITE_CREATOR)
+@require_http_methods(["GET", "POST"])
+def new_job(request):
+    sis_account_id = request.LTI["custom_canvas_account_sis_id"]
+    terms, _current_term_id = get_term_data_for_school(sis_account_id)
+    school_id = sis_account_id.split(":")[1]
+    canvas_site_templates = get_canvas_site_templates_for_school(school_id)
+    potential_course_sites_query = None
+    departments = []
+    course_groups = []
+
+    # Only display the Course Groups dropdown if the tool is launched in the COLGSAS sub-account
+    if school_id == 'colgsas':
+        try:
+            course_groups = get_course_group_data_for_school(sis_account_id)
+        except Exception:
+            logger.exception(f"Failed to get course groups with sis_account_id {sis_account_id}")
+    # For all other schools, display just the Departments dropdown
+    else:
+        try:
+            departments = get_department_data_for_school(sis_account_id)
+        except Exception:
+            logger.exception(f"Failed to get departments with sis_account_id {sis_account_id}")
+
+    if request.method == "POST":
+        selected_term_id = request.POST["course-term"]
+        # Retrieve all course instances for the given term_id and account that do not have Canvas course sites
+        # nor are set to be fed into Canvas via the automated feed
+        potential_course_sites_query = get_course_instance_query_set(
+            selected_term_id, sis_account_id
+        ).filter(canvas_course_id__isnull=True, sync_to_canvas=0)
+
+    # TODO maybe better to use template tag unless used elsewhere?
+    potential_course_site_count = (
+        potential_course_sites_query.count() if potential_course_sites_query else 0
+    )
+
+    context = {
+        "terms": terms,
+        "potential_course_sites": potential_course_sites_query,
+        "potential_site_count": potential_course_site_count,
+        "canvas_site_templates": canvas_site_templates,
+        "departments": departments,
+        "course_groups": course_groups
+    }
+
+    return render(request, "bulk_site_creator/new_job.html", context=context)
+
+
+@login_required
+@lti_permission_required(settings.PERMISSION_BULK_SITE_CREATOR)
+@require_http_methods(["POST"])
+def create_bulk_job(request: HttpRequest) -> Optional[JsonResponse]:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 58 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 99% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 198-252 into a separate function.